### ISSUE TITLE
Support shuffle operation

### ISSUE THREAD

**Issue by gwsshs22** [snuspl/cruise#40]
Dolphin should support shuffle operations to avoid redundant scatter-gather operations to exchange data among computeTasks. We can use cay/services/shuffle submodule, which has simple primitives of push style shuffle, to add shuffle APIs to dolphin after #36 is merged. 

After the issue is resolved, we should make an another issue to apply the shuffle API to existing dolphin examples (e.g., SVD, ALS) to improve performance.
